setup.sh

```diff
@@ -191,25 +191,37 @@ function update() {
 }
 
 # Reset Develop or Stable branch
+function check_git_changes() {
+    if [ -z "$(git status --porcelain)" ]; then
+        echo "No changes in git directory"
+        return 0
+    else
+        echo "Changes in git directory"
+        return 1
+    fi
+}
 function reset() {
     echo_block "Resetting branch and virtual env"
 
     if [ "1" == $(git branch -vv |grep -cE "\* develop|\* stable") ]
     then
-
-        read -p "Reset git branch? (This will remove all changes you made!) [y/N]? "
-        if [[ $REPLY =~ ^[Yy]$ ]]; then
-
-            git fetch -a
-
-            if [ "1" == $(git branch -vv | grep -c "* develop") ]
-            then
-                echo "- Hard resetting of 'develop' branch."
-                git reset --hard origin/develop
-            elif [ "1" == $(git branch -vv | grep -c "* stable") ]
-            then
-                echo "- Hard resetting of 'stable' branch."
-                git reset --hard origin/stable
+        if check_git_changes; then
```

Comment:
unfortunately,  this change ain't working .

this is bash - you can't assume that "1" is truthy, and 0 is falsy.
Running this script shows:

```
----------------------------
Resetting branch and virtual env
----------------------------
Changes in git directory
- Deleting your previous virtual env

```


It clearly detected "changes in git directory" - but it ignored all changes - as the check is invalid.

Reply:
damn you chatgpt :D

it should work, should according to SO as well: https://stackoverflow.com/questions/5431909/returning-a-boolean-from-a-bash-function

Not a bash hero, so if you got any ideas, feel free to pass them along

Reply:
quite honestly, i'd have believed that as well by just looking at the code - but after testing - i saw it ain't working (i'm always skeptical about bash comparisons ...).

Reply:
`0 is true` and `1 is false` in bash, btw. 

because it's analogue to the exit value of the process. just remembered my unix classes

Reply:
fixed it now

Reply:
Works now :+1: